### ISSUE TITLE
Extend F expressions support

### ISSUE THREAD

**Issue by Stranger6667** [django-money/django-money#167]
This code works:
```python
instance = ModelWithVanillaMoneyField.objects.create(money=Money(100, 'USD'))
instance.money = F('money') + Money(100, 'USD')
instance.save()
```
Same for subtraction.
Multiplication and division is not supported yet, so this code fails:
```python
instance = ModelWithVanillaMoneyField.objects.create(money=Money(100, 'USD'))
instance.money = F('money') * 2
instance.save()
```
With
```python
    def _check_currency(obj, field, amt):
        currency_field_name = get_currency_field_name(field.name)
>       if obj.__dict__[currency_field_name] != str(amt.currency):
E       AttributeError: 'int' object has no attribute 'currency'

djmoney/models/fields.py:208: AttributeError
```
I'll create new PR for this issue after #165 will be merged to master (I hope it will :))
